web/modules/custom/joinup_core/joinup_core.post_update.php

```diff
@@ -90,3 +94,272 @@ function joinup_core_post_update_remove_action_transfer_solution_ownership() {
+      'entity_id' => $id,
+    ]);
+
+    foreach ($memberships as $key => $membership) {
+      unset($memberships[$key]);
+      // Prepend "uid:" to each key to avoid conflicts with membership ids.
+      $memberships['uid:' . $membership->getOwnerId()] = $membership;
+    }
+    return $memberships;
+  };
+
+  foreach (_joinup_core_get_duplicated_ids() as $original_id => $duplicate_ids) {
+    $group = Rdf::load($original_id);
+    // In case the original entity is not found, it means that the entity with
+    // the duplicate id was kept after the migration.
+    if (empty($group)) {
+      continue;
+    }
+
+    /** @var \Drupal\og\OgMembershipInterface[] $memberships */
+    $memberships = $memberships_by_uid($original_id);
+    foreach ($duplicate_ids as $duplicate_id) {
+      /** @var \Drupal\og\OgMembershipInterface[] $duplicate_memberships */
+      $duplicate_memberships = $memberships_by_uid($duplicate_id);
+      foreach ($duplicate_memberships as $key => $duplicate_membership) {
+        // If the membership does not exist in the original entity, transfer it
+        // and maintain the values.
+        if (!isset($memberships[$key])) {
+          $duplicate_membership->setGroup($group);
+          $duplicate_membership->skip_notification = 1;
+          $duplicate_membership->save();
+        }
+        // If the membership does exist in the original entity, add the roles to
+        // the existing membership. We do not need to filter anything out since
+        // og is already handling it in ::setRoles.
+        // @see \Drupal\og\OgMembershipInterface::getRoles.
+        else {
+          $duplicate_roles = $duplicate_memberships[$key]->getRoles();
+          $original_membership = $memberships[$key];
+          $original_membership->setRoles($duplicate_roles);
+          $original_membership->skip_notification = 1;
+          $original_membership->save();
+        }
+      }
+    }
+  }
+}
+
+/**
+ * Move content from duplicates to the original entity.
+ */
+function _joinup_core_post_update_ctt_duplicates_merge_content() {
+  /** @var \Drupal\og\MembershipManagerInterface $membership_manager */
+  $membership_manager = \Drupal::service('og.membership_manager');
+  foreach (_joinup_core_get_duplicated_ids() as $original_id => $duplicates_ids) {
+    $original_group = Rdf::load($original_id);
+    if (empty($original_group)) {
+      continue;
+    }
+
+    foreach ($duplicates_ids as $duplicate_id) {
+      $duplicate_group = Rdf::load($duplicate_id);
+      if (empty($duplicate_group)) {
+        continue;
+      }
+
+      // Unset the relationship to any distribution so that they can properly
+      // moved.
+      $duplicate_group->set('field_is_distribution', NULL);
+      $duplicate_group->skip_notification = 1;
+      $duplicate_group->save();
+
+      $content_ids = $membership_manager->getGroupContentIds($duplicate_group);
+      foreach ($content_ids as $entity_type => $entity_ids) {
```

Comment:
Let's use the good naming practices from the core `$entity_type_id` (`$entity_type` refers to the object).